timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/rdf/Collection.java

```diff
@@ -79,6 +79,15 @@ public void add(Vertex entityVertex) {
     }
   }
 
+  public void add(Vertex entityVertex, Collection existingCollection) {
```

Comment:
I know. But I respectfully disagree with them. Also they're not saying you must never do it. Just that passing null is sometimes faster. So I suggest you just disable that warning.

See: https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained and http://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments So. If you have a function that does one thing when the argument is null, and another when the argument is non-null you might better write two functions (point 2 in the SO answer). But if you don't want that to cascade throughout your codebase. So all methods one layer above that function might just as well receive an optional as a parameter. Unless they're called really really often in that case calling them with `null` might be faster.